docs/topics/feed-exports.rst

```diff
@@ -267,10 +267,137 @@ once all the file contents have been written (i.e. at the end of the crawl) is
 that file uploaded to the feed URI.
 
 If you want item delivery to start earlier when using one of these storage
-backends, use :setting:`FEED_EXPORT_BATCH_ITEM_COUNT` to split the output items
-in multiple files, with the specified maximum item count per file. That way, as
-soon as a file reaches the maximum item count, that file is delivered to the
-feed URI, allowing item delivery to start way before the end of the crawl.
+backends, use :ref:`batches` to split the output items into multiple
+files, delivered sequentially after a given criteria is met, e.g. after each
+file contains a given number of items or after a given time has passed since
+the last file delivered.
+
+.. _batches:
+
+Batches
+=======
+
+.. versionadded:: VERSION
+
+Batches enable multiple sequential files to be generated for a feed instead of getting
+a single file. This is useful in getting item deliveries earlier before the crawl has
+been ended.
+
+Batching is controlled by a batch handler, which is defined through the 
+:ref:`batch feed option <feed-options>`.
+For information about enabling and configuring the default batch handler, see
+:class:`~scrapy.extensions.batches.BatchHandler`. You can alternatively 
+:ref:`create your own batch handler <custom_batch_handler>`.
+
+If you enable batching, you must also set :ref:`feed URIs appropriately<batch-uri>` with placeholders.
+
+.. _default_batch_handler:
+
+Default Batch Handler
+----------------------
+
+.. autoclass:: scrapy.extensions.batches.BatchHandler
+
+.. _custom_batch_handler:
+
+Custom Batch Handler
+--------------------
+
+Each batch handler is a class that must implement the following methods:
+
+.. method:: __init__(self, feed_options)
+
+    Initialize the batch handler.
+
+    Parameters:
+
+    :param feed_options: feed-specific :ref:`options <feed-options>`
+    :type feed_options: :class:`dict`
+
+    Required attributes:
+
+    * `self.batch_id` (:class:`int`): id number of current batch, must be initialised to 0.
+    * `self.enabled` (:class:`bool`): must be `True` if a non-empty limit is passed, otherwise `False`.
+
+.. method:: item_added(self)
+
+    Update batch state info. Will be called everytime an item is exported to feed.
```

Comment:
Moving internal data updating code to `should_trigger` will make `item_added` method useless. I guess I should remove `item_added` if we are doing all the updating in `should_trigger`?

Reply:
I’m slightly inclined to keep it because I think it has some potential for extension in the future. I’m thinking that in the future (or maybe now?) we could add an `item` parameter to it, so that a batch handler can trigger batch switches based on item content.

I’m also thinking that we might want to eventually have a counterpart `before_item_added` method, which would be called before an item is added to a batch and get not only the `item` as a parameter, but possibly also the `data` that will be written into the file, as both things could also be used by a batch handler to determine whether or not to switch to a new batch.

Instead of removing `item_added`, maybe we could remove `should_trigger` and have `item_adapter` work the way `should_trigger` does, so instead of:

```python
self.batches[slot.uri_template].item_added()
if self.batches[slot.uri_template].should_trigger():
    ...
```

we could have:

```python
create_new_batch = self.batches[slot.uri_template].item_added()
if create_new_batch:
    ...
```

Reply:
Oh nice, this could also be helpful in implementing hard limit for file size.